pkg/events/trigger/context.go

```diff
@@ -1,72 +1,83 @@
 package trigger
 
 import (
+	"fmt"
 	"sync"
 
 	"github.com/aquasecurity/tracee/pkg/counter"
 	"github.com/aquasecurity/tracee/pkg/errfmt"
 	"github.com/aquasecurity/tracee/pkg/events/parse"
+	"github.com/aquasecurity/tracee/pkg/logger"
 	"github.com/aquasecurity/tracee/types/trace"
 )
 
 const ContextArgName = "caller_context_id"
 
+// Context is an interface for a struct used to store the triggering events context.
 type Context interface {
 	Store(trace.Event) uint64               // store an invoke context
-	Get(uint64) (trace.Event, bool)         // get an invoked event context
+	Load(uint64) (trace.Event, bool)        // loads an invoked event context
 	Apply(trace.Event) (trace.Event, error) // apply an invoked event context (implicitly gets the event)
 }
 
-// context is a struct used to store the triggering events context
 type context struct {
 	store   map[uint64]trace.Event
-	mutex   sync.RWMutex
+	mutex   *sync.Mutex
 	counter counter.Counter
 }
 
+// NewContext creates a new context store.
 func NewContext() *context {
 	return &context{
 		store:   make(map[uint64]trace.Event),
-		mutex:   sync.RWMutex{},
-		counter: counter.NewCounter(1),
```

Comment:
FYI, pretty sure I left it at one for a reason (my bad for not documenting it).

Reply:
Thats because there was a "read and increment":

```diff
        id := uint64(store.counter.Read())
-       _ = store.counter.Increment()
+       _ = store.counter.Increase()
```

so you would pick 1 and increment to two. Now, its 0 but at the first Load() it is 1 (because of the atomic increment returning the new value). So its the same thing as before, but it starts at 0 due to a single atomic operation and not two.